crypto/storage/client.go

```diff
@@ -0,0 +1,109 @@
+package storage
+
+import (
+	"context"
+	"crypto"
+	"fmt"
+	"github.com/nuts-foundation/nuts-node/crypto/storage/httpclient"
+	"github.com/nuts-foundation/nuts-node/crypto/util"
+	"net/http"
+)
+
+// APIClient implements the Storage interface. It uses a simple HTTP protocol to connect to an external storage server.
+// This server can either be a secret store itself, or proxy the request to a key store such as Hashicorp Vault or Azure Key Vault.
+// It allows us to keep the codebase clean and allow other parties to write their own adaptor.
+type APIClient struct {
```

Comment:
general comment: allow `context.Context` to be passed to allow cancellation (usage of `context.Background` indicates missing context passing).

Reply:
I Agree, but that changes the whole crypto.Storage interface. Perhaps part of a separate PR?

Reply:
The ListKeys is also a bit weird since it doesn't return errors.